src/Compilers/CSharp/Portable/FlowAnalysis/ExpressionTreeRefLikeWalker.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.FlowAnalysis
+{
+    internal class ExpressionTreeRefLikeWalker : BoundTreeWalker
```

Comment:
Why can't we add this validation to existing walkers of expression trees?

Reply:
I think it makes more sense to pull this into the existing walker--should I just make this check into some private helper and call it from the various places that expressions are visited in `DiagnosticsPass_ExpressionTrees`?